src/dev/build/tasks/clean_tasks.js

```diff
@@ -57,13 +59,94 @@ export const CleanExtraFilesFromModulesTask = {
   description: 'Cleaning tests, examples, docs, etc. from node_modules',
 
   async run(config, log, build) {
-    await deleteAll(log, [
-      build.resolvePath('node_modules/**/test/**/*'),
-      build.resolvePath('node_modules/**/tests/**/*'),
-      build.resolvePath('node_modules/**/example/**/*'),
-      build.resolvePath('node_modules/**/examples/**/*'),
-      build.resolvePath('node_modules/**/.bin/**/*'),
-    ]);
+    const deleteFromNodeModules = globs => {
+      return deleteAll(
+        log,
+        globs.map(p => build.resolvePath(`node_modules/**/${p}`))
+      );
+    };
+
+    const tests = [
+      'test',
+      'tests',
+      '__tests__',
+      'mocha.opts',
+      '*.test.js',
+      '*.snap',
+      'coverage',
+    ];
+    const docs = [
+      'doc',
+      'docs',
+      'CONTRIBUTING.md',
+      'Contributing.md',
+      'contributing.md',
+      'History.md',
+      'HISTORY.md',
+      'history.md',
+      'CHANGELOG.md',
+      'Changelog.md',
+      'changelog.md',
+    ];
+    const examples = ['example', 'examples', 'demo', 'samples'];
+    const bins = ['.bin'];
+    const linters = [
+      '.eslintrc',
+      '.eslintrc.js',
+      '.eslintrc.yml',
+      '.prettierrc',
+      '.jshintrc',
+      '.babelrc',
+      '.jscs.json',
+      '.lint',
+    ];
+    const hints = ['*.flow', '*.webidl', '*.map'];
+    const scripts = [
+      '*.sh',
+      '*.bat',
+      '*.exe',
+      'Gruntfile.js',
+      'gulpfile.js',
+      'Makefile',
+    ];
+    const transpiledSources = ['*.coffee', '*.scss', '*.sass'];
```

Comment:
nit: The are the opposite of transpiled sources. They are untranspiled sources, or perhaps transpilable sources.

Reply:
Should we add .ts and .tsx to the mix since those can't be transpiled on the fly?

Reply:
👍 Updated both with 190f60935